File: cpp/program/Clothing.cpp
#include <iostream>
#include <string>
#include "Product.cpp"

using namespace std;

class Clothing : public Product {
private:
    string size;
    string material;
    string gender;

public:
    Clothing(string idProduct, string name, string brand, double price, string size, string material, string gender)
    : Product(idProduct, name, brand, price), size(size), material(material), gender(gender) {
    }

    string getSize() const {
        return size;
    }

    string getMaterial() const {
        return material;
    }

    string getGender() const {
        return gender;
    }

    void setSize(const string& sz) {
        size = sz;
    }

    void setMaterial(const string& mt) {
        material = mt;
    }

    void setGender(const string& gd) {
        gender = gd;
    }

    ~Clothing() {
    }
};

File: php/program/Clothing.php
<?php

require_once("Product.php");

class Clothing extends Product {
    protected $size;
    protected $material;
    protected $gender;

    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender) {
        parent::__construct($idProduct, $name, $brand, $price);
        $this->size = $size;
        $this->material = $material;
        $this->gender = $gender;
    }

    public function getSize() {
        return $this->size;
    }

    public function getMaterial() {
        return $this->material;
    }

    public function getGender() {
        return $this->gender;
    }

    public function setSize($size) {
        $this->size = $size;
    }

    public function setMaterial($material) {
        $this->material = $material;
    }

    public function setGender($gender) {
        $this->gender = $gender;
    }
}

?>
